tests/spi/spi.go

```diff
@@ -0,0 +1,241 @@
+package spi
+
+import (
+	"time"
+
+	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const ()
+
+var ()
+
+/*
+ * Component: spi
+ * Description: Contains tests covering basic spi scenarios
+ */
+
+var _ = framework.SPISuiteDescribe(Label("spi-suite"), func() {
+
+	defer GinkgoRecover()
+
+	var fw *framework.Framework
+	var err error
+	var namespace string
+
+	BeforeAll(func() {
+		// Initialize the tests controllers
+		fw, err = framework.NewFramework(utils.GetGeneratedNamespace("spi-demos"))
+		Expect(err).NotTo(HaveOccurred())
+		namespace = fw.UserNamespace
+		Expect(namespace).NotTo(BeEmpty())
+	})
+
+	Describe("SVPI-398 - Token upload rest endpoint", Ordered, func() {
+
+		// Clean up after running these tests and before the next tests block: can't have multiple AccessTokens in Injected phase
+		AfterAll(func() {
```

Comment:
I think we can delete it if the test passes. This way, we can dig better when it fails. Same on line 202. WDYT?

We can add `if !CurrentSpecReport().Failed() {`

Reply:
yes, it makes perfect sense. adding it! thanks!